src/hasher512.cpp

```diff
@@ -160,9 +160,9 @@ namespace libtorrent
 #endif
 		}
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), data.size());
+		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), std::uint32_t(data.size()));
```

Comment:
hmmm...the signature from openssl is
`int SHA512_Update(SHA512_CTX *c, const void *data, size_t len);`
what version of openssl are you using?